test/besselk_test.jl

```diff
@@ -67,6 +67,12 @@ k1x_32 = besselk1x.(Float32.(x))
 @test besselk(100, 3.9) ≈ SpecialFunctions.besselk(100, 3.9)
 @test besselk(100, 234.0) ≈ SpecialFunctions.besselk(100, 234.0)
 
+# test half-integer orders:
+for v in (-3/2, -1/2, 1/2, 3/2, 5/2, 7/2, 9/2)
```

Comment:
I think testing for more larger orders and for more x values will be beneficial. These tests are so cheap and if we add other branches it will be nice to test around and after the cutoff.